### PR TITLE
feat(replays): Replay layout timeline size consistent when full screening

### LINKED ISSUE #59005: [Replay Details] Default Timeline zoom level to 100%
Also we should make sure that the zoom buttons are disabled as needed when we're at the min or max zoom level.
Dead clicks need not be detected!

### PR DESCRIPTION
**Pull request by c298lee**
The timeline size was resetting to the default size when full screened, so moved the timeline states to replay context to keep size consistent. The default size has also been changed to 1x, with the largest size being the amount of minutes a replay is

Relates to https://github.com/getsentry/team-replay/issues/199
Closes https://github.com/getsentry/sentry/issues/59005

### CODE REVIEW

File: static/app/components/replays/replayContext.tsx [hunk 1]
```diff
@@ -114,11 +114,21 @@ interface ReplayPlayerContextProps extends HighlightCallbacks {
    */
   setSpeed: (speed: number) => void;
 
+  /**
+   * Set the timeline to the specific size
+   */
+  setTimelineSize: (size: number) => void;
```
Comment (ryan953) [inline static/app/components/replays/replayContext.tsx:120]: Nit: The comment below does this a bit, but can we update the names even more to indicate that the value is a percentage?

```suggestion
  /**
   * Set the timeline width to the specific scale, starting at 100% and growing larger
   */
  setTimelineScale: (percent: number) => void;
```

File: static/app/components/replays/replayContext.tsx [hunk 2]
```diff
@@ -114,11 +114,21 @@ interface ReplayPlayerContextProps extends HighlightCallbacks {
...
+
   /**
    * The speed for normal playback
    */
   speed: number;
 
+  /**
+   * Size of timeline as a percentage of the original size
+   */
+  timelineSize: number;
```
Comment (ryan953) [inline static/app/components/replays/replayContext.tsx:130]:
```suggestion
  timelineScale: number;
```

File: static/app/components/replays/breadcrumbs/replayTimeline.tsx [hunk 3]
```diff
@@ -56,7 +59,7 @@ function ReplayTimeline({size}: Props) {
     <VisiblePanel ref={panelRef} {...mouseTrackingProps}>
       <Stacked
         style={{
-          width: `${size}%`,
+          width: `${timelineSize}%`,
           transform: `translate(${translatePercentage}, 0%)`,
```
Comment (ryan953) [inline static/app/components/replays/breadcrumbs/replayTimeline.tsx:63]: nit:
I'd call `toPercent` on both of these lines, which leaves the values as numbers for as long as possible:
```suggestion
          width: toPercent(timelineSize),
          transform: `translate(${toPercent(translatePercentage)}, 0%)`,
```

File: static/app/components/replays/replayController.tsx [hunk 4]
```diff
@@ -146,24 +146,27 @@ function ReplayOptionsMenu({speedOptions}: {speedOptions: number[]}) {
...
     <ButtonBar merged>
       <Button
         size="xs"
         title={t('Zoom out')}
         icon={<IconSubtract size="xs" />}
         borderless
-        onClick={() => setSize(Math.max(size - 50, 100))}
+        onClick={() => setTimelineSize(Math.max(timelineSize - 50, 100))}
         aria-label={t('Zoom out')}
+        disabled={timelineSize === 100 ? true : false}
```
Comment (ryan953) [inline static/app/components/replays/replayController.tsx:160]: The left side of the ternary returns a boolean, no need for the `true` and `false` keywords :D

```suggestion
        disabled={timelineSize === 100}
```

File: static/app/components/replays/replayController.tsx [hunk 5]
```diff
@@ -146,24 +146,27 @@ function ReplayOptionsMenu({speedOptions}: {speedOptions: number[]}) {
...
       />
       <Button
         size="xs"
         title={t('Zoom in')}
         icon={<IconAdd size="xs" />}
         borderless
-        onClick={() => setSize(Math.min(size + 50, 1000))}
+        onClick={() => setTimelineSize(Math.min(timelineSize + 50, 1000))}
         aria-label={t('Zoom in')}
+        disabled={timelineSize === 1000 ? true : false}
```
Comment (ryan953) [inline static/app/components/replays/replayController.tsx:169]:
```suggestion
        disabled={timelineSize === 1000}
```

**Review by ryan953 — APPROVED** [review]

File: static/app/components/replays/replayContext.tsx [hunk 6]
```diff
@@ -114,11 +114,21 @@ interface ReplayPlayerContextProps extends HighlightCallbacks {
...
+  setTimelineScale: (size: number) => void;
+
   /**
    * The speed for normal playback
    */
   speed: number;
 
+  /**
+   * Scale of the timeline width
+   */
```
Comment (billyvg) [inline static/app/components/replays/replayContext.tsx:129]: Might be helpful to give an expected range, i.e. is this `0 < x <= 1`?

**Review by billyvg — APPROVED** [review]

**Pull request merged**